src/idom/utils.py

```diff
@@ -63,80 +67,149 @@ def html_to_vdom(source: str, *transforms: _ModelTransform) -> Dict[str, Any]:
             dictionary which will be replaced by ``new``. For example, you could use a
             transform function to add highlighting to a ``<code/>`` block.
     """
-    parser = HtmlParser()
-    parser.feed(source)
-    root = parser.model()
-    to_visit = [root]
-    while to_visit:
-        node = to_visit.pop(0)
-        if isinstance(node, dict) and "children" in node:
-            transformed = []
-            for child in node["children"]:
-                if isinstance(child, dict):
-                    for t in transforms:
-                        child = t(child)
-                if child is not None:
-                    transformed.append(child)
-                    to_visit.append(child)
-            node["children"] = transformed
-            if "attributes" in node and not node["attributes"]:
-                del node["attributes"]
-            if "children" in node and not node["children"]:
-                del node["children"]
-    return root
-
-
-class HtmlParser(_HTMLParser):
-    """HTML to VDOM parser
-
-    Example:
-
-        .. code-block::
-
-            parser = HtmlParser()
-
-            parser.feed(an_html_string)
-            parser.feed(another_html_string)
-            ...
-
-            vdom = parser.model()
+    if not isinstance(html, str):
+        raise TypeError(f"Encountered unsupported type {type(html)} from {html}")
```

Comment:
```suggestion
        raise TypeError(f"Expected html to be a string, not {type(html).__name__}")
```

Reply:
Calling `__name__` on a `type` call is risky, since it does not handle `None`

Reply:
It seems fairly reliable to say it's present and a string since the constructor for custom classes raises when `type(None, (object,), {})` because the new class' name must be a string. I don't think I've ever seen a class without a `__name__` before. Do you know of any examples?

Reply:
Also `type(None).__name__ == "NoneType"`

Reply:
True about the `type(None).__name__`. I was thinking of `None.__name__`, which is an exception.